src/server/server.h

```diff
@@ -319,4 +359,8 @@ class Server {
   std::atomic<size_t> watched_key_size_ = 0;
   std::map<std::string, std::set<redis::Connection *>> watched_key_map_;
   std::shared_mutex watched_key_mutex_;
+
+  // SCAN ring buffer
+  std::atomic<uint16_t> cursor_counter_ = {0};
+  std::array<CursorDictElement, CURSOR_DICT_SIZE> cursor_dict_;
```

Comment:
@PragmaTwice would a `sizeof(CursorDictElement) * CURSOR_DICT_SIZE` dict in `Server` makes it too large? And If `Server` is large, would it matters?

Reply:
`sizeof(cursor_dict_)` =  40 * 1024 * 16 = 640KB.
Are there any issues with a large `Server`?

Reply:
use a unordered_map?

Reply:
@jihuayu I guess 640KB is small, using an sequence of memory is ok. But I don't know do we need to copy or do something with `Server`, if we need, the `Server` would become an large object. So I ask for Twice's idea

Reply:
Seems the global server object is allocated directly on the stack rather than heap, so I think it is better to put the array on heap, e.g. `std::vector` or `std::unique_ptr<T[]>`.

Reply:
OK, I have changed `std::array` to `std::unique_ptr<std::array>`